src/webui/webapplication.cpp

```diff
@@ -527,11 +528,14 @@ void WebApplication::sessionStart()
 
     // remove outdated sessions
     const qint64 now = QDateTime::currentMSecsSinceEpoch() / 1000;
-    const QHash<QString, WebSession *> sessionsCopy {m_sessions};
-    for (const auto session : sessionsCopy) {
-        if ((now - session->timestamp()) > INACTIVE_TIME)
-            delete m_sessions.take(session->id());
-    }
+    Algorithm::removeIf(m_sessions, [now](const QString &, const WebSession *session)
+    {
+        if ((now - session->timestamp()) <= INACTIVE_TIME)
+            return false;
+
+        delete session;
```

Comment:
Am I the only one confused that the predicate is changing something, not just testing some condition?

Reply:
Obviously I'm biased, so it is acceptable to me.
One can argue that `BinaryPredicate` is not a good name for this kind of usage since it doesn't align with C++ named requirements, but I can't think of a better name... and also that requirement is for std lib which we are not...

Reply:
>Am I the only one confused that the predicate is changing something, not just testing some condition?

Or change to `QHash<QString, std::unique_ptr<WebSession>> m_sessions;`...
"All problems in computer science can be solved by another level of indirection."

Reply:
>Perhaps we can just relax/assume that here the predicate concept does not impose immutability on container elements?

Yes, but why? Personally, I don't see any improvements from this code in places like this, so I would prefer the old code.

>Or change to `QHash<QString, std::unique_ptr<WebSession>> m_sessions;`...

I thought about it too. At least it would make use `removeIf` justified.

Reply:
>>Perhaps we can just relax/assume that here the predicate concept does not impose immutability on container elements?
>
>Yes, but why? 

So we either duplicate the guts of `removeIf` or go back to `QMutableHashIterator`, both are not as succinct as the current one here I think.

>>Or change to QHash<QString, std::unique_ptr<WebSession>> m_sessions;...
>
>I thought about it too. At least it would make use removeIf justified.

There are some problems using unique_ptr with qhash so that way is blocked.

Reply:
>There are some problems using unique_ptr with qhash so that way is blocked.

What's the problem? Apparently you hust need to provide qHash function template for unique_ptr?..

Reply:
>What's the problem? 

Seems to be this one: https://bugreports.qt.io/browse/QTBUG-59188
At `m_sessions[m_currentSession->id()] = m_currentSession;` statement.

Reply:
Okay, well, you can merge it if you want, although I'm still confused by the use of predicates. It's not so much the name "predicate", but the fact that now the operations of deleting an element and deleting an object are at a distance from each other in the code (although logically it is single operation).